# HG changeset patch
# User Rik <rik@octave.org>
# Date 1519688608 28800
#      Mon Feb 26 15:43:28 2018 -0800
# Node ID b901d3123745085cfe651b7e3fb0168945d5b268
# Parent  b0be07092078260e4e25301843b5f02ef5ffe7b3
# Parent  d3a79cb734d2ca32f2a4ca22fc4d368037cbbca0
maint: merge stable to default.

diff --git a/liboctave/array/CSparse.cc b/liboctave/array/CSparse.cc
--- a/liboctave/array/CSparse.cc
+++ b/liboctave/array/CSparse.cc
@@ -4706,19 +4706,20 @@ SparseComplexMatrix::bsolve (MatrixType&
                       for (octave_idx_type i = 0; i < b_nr; i++)
                         {
                           Complex tmp = Bx[i];
                           if (tmp != 0.0)
                             {
                               if (ii == x_nz)
                                 {
                                   // Resize the sparse matrix
-                                  octave_idx_type sz = x_nz *
-                                                       (b_nc - j) / b_nc;
-                                  sz = (sz > 10 ? sz : 10) + x_nz;
+                                  octave_idx_type sz;
+                                  sz = (static_cast<double> (b_nc) - j) / b_nc
+                                       * x_nz;
+                                  sz = x_nz + (sz > 100 ? sz : 100);
                                   retval.change_capacity (sz);
                                   x_nz = sz;
                                 }
                               retval.xdata (ii) = tmp;
                               retval.xridx (ii++) = i;
                             }
                         }
                       retval.xcidx (j+1) = ii;
@@ -6147,18 +6148,20 @@ SparseComplexMatrix::fsolve (MatrixType&
                   for (octave_idx_type i = 0; i < b_nr; i++)
                     {
                       Complex tmp = Xx[i];
                       if (tmp != 0.0)
                         {
                           if (ii == x_nz)
                             {
                               // Resize the sparse matrix
-                              octave_idx_type sz = x_nz * (b_nc - j) / b_nc;
-                              sz = (sz > 10 ? sz : 10) + x_nz;
+                              octave_idx_type sz;
+                              sz = (static_cast<double> (b_nc) - j) / b_nc
+                                   * x_nz;
+                              sz = x_nz + (sz > 100 ? sz : 100);
                               retval.change_capacity (sz);
                               x_nz = sz;
                             }
                           retval.xdata (ii) = tmp;
                           retval.xridx (ii++) = i;
                         }
                     }
                   retval.xcidx (j+1) = ii;
@@ -6651,18 +6654,20 @@ SparseComplexMatrix::fsolve (MatrixType&
                   for (octave_idx_type i = 0; i < b_nr; i++)
                     {
                       Complex tmp = Xx[i];
                       if (tmp != 0.0)
                         {
                           if (ii == x_nz)
                             {
                               // Resize the sparse matrix
-                              octave_idx_type sz = x_nz * (b_nc - j) / b_nc;
-                              sz = (sz > 10 ? sz : 10) + x_nz;
+                              octave_idx_type sz;
+                              sz = (static_cast<double> (b_nc) - j) / b_nc
+                                   * x_nz;
+                              sz = x_nz + (sz > 100 ? sz : 100);
                               retval.change_capacity (sz);
                               x_nz = sz;
                             }
                           retval.xdata (ii) = tmp;
                           retval.xridx (ii++) = i;
                         }
                     }
                   retval.xcidx (j+1) = ii;
